src/benchmark/tpcc_benchmark.cpp

```diff
@@ -0,0 +1,320 @@
+    Assert(order_line_table &&
+               order_line_table->row_count() == static_cast<size_t>(num_warehouses * NUM_DISTRICTS_PER_WAREHOUSE),
+           "Did not find COUNT(*) FROM ORDER_LINE for all districts");
+
+    for (auto row_id = size_t{0}; row_id < order_line_table->row_count(); ++row_id) {
+      Assert(order_table->get_value<int64_t>(ColumnID{2}, row_id) ==
+                 order_line_table->get_value<int64_t>(ColumnID{2}, row_id),
+             "Mismatching order_line count");
+    }
+  }
+
+  {
+    std::cout << "  -> Running consistency check 5" << std::endl;
+    auto pipeline =
+        SQLPipelineBuilder{
+            "SELECT * FROM \"ORDER\" WHERE O_CARRIER_ID = -1 AND NOT EXISTS (SELECT NO_W_ID FROM NEW_ORDER WHERE "
+            "O_W_ID = NO_W_ID AND O_D_ID = NO_D_ID AND NO_O_ID = O_ID)"}
+            .create_pipeline();
+    const auto [pipeline_status, table] = pipeline.get_result_table();
+    Assert(table && table->row_count() == size_t{0}, "Found fulfilled order without O_CARRIER_ID");
+  }
+
+  {
+    std::cout << "  -> Running consistency check 6" << std::endl;
+    auto pipeline =
+        SQLPipelineBuilder{
+            "SELECT O_W_ID, O_D_ID, O_ID, MAX(O_OL_CNT), COUNT(*) FROM \"ORDER\" LEFT JOIN ORDER_LINE ON O_W_ID = "
+            "OL_W_ID AND O_D_ID = OL_D_ID AND O_ID = OL_O_ID GROUP BY O_W_ID, O_D_ID, O_ID"}
+            .create_pipeline();
+    const auto [pipeline_status, table] = pipeline.get_result_table();
+    Assert(table && table->row_count() > size_t{0}, "Failed to retrieve order / order lines");
+    for (auto row_id = size_t{0}; row_id < table->row_count(); ++row_id) {
+      Assert(table->get_value<int32_t>(ColumnID{3}, row_id) == table->get_value<int64_t>(ColumnID{4}, row_id),
+             "Mismatching number of order lines");
+    }
+  }
+
+  {
+    std::cout << "  -> Running consistency check 7" << std::endl;
+    auto pipeline =
+        SQLPipelineBuilder{
+            "SELECT * FROM ORDER_LINE LEFT JOIN \"ORDER\" ON OL_W_ID = O_W_ID AND OL_D_ID = O_D_ID AND OL_O_ID = O_ID "
+            "WHERE OL_DELIVERY_D = -1 AND O_CARRIER_ID <> -1"}
+            .create_pipeline();
+    const auto [pipeline_status, table] = pipeline.get_result_table();
+    Assert(table && table->row_count() == size_t{0},
+           "Found order line without OL_DELIVERY_D even though the order was delivered");
+  }
+
+  {
+    std::cout << "  -> Running consistency check 8" << std::endl;
+    auto pipeline =
+        SQLPipelineBuilder{
+            "SELECT W_ID, MAX(W_YTD), SUM(H_AMOUNT) FROM WAREHOUSE, HISTORY WHERE W_ID = H_W_ID GROUP BY W_ID"}
+            .create_pipeline();
+    const auto [pipeline_status, table] = pipeline.get_result_table();
+    Assert(table && table->row_count() == static_cast<size_t>(num_warehouses), "Lost a warehouse");
+    for (auto row_id = size_t{0}; row_id < table->row_count(); ++row_id) {
+      const auto w_ytd = double{table->get_value<float>(ColumnID{1}, row_id)};
```

Comment:
The table holds a float. If you call `get_value<double>` on a float column, it fails.

Reply:
Remember that we need to resolve the segment types. Just because `float` is convertible to `double` doesn't mean that `ValueSegment<float>` is convertible to `ValueSegment<double>`